message_ix_models/model/transport/roadmap.py

```diff
@@ -0,0 +1,176 @@
+    Niger, Nigeria, Rwanda, Sudan, Senegal, Sierra Leone, Somalia, Sao Tome and
+    Principe, Swaziland, Seychelles, Chad, Togo, Tunisia, United Republic of Tanzania,
+    Uganda, South Africa, Zambia, Zimbabwe -- see sheet **Countries by Roadmap
+    Region** in RoadmapResults_2017.xlsx.
+"""
+
+import pandas as pd
+from plotnine import save_as_pdf_pages
+
+from message_ix_models.util import package_data_path
+
+#: Name of the file containing the data.
+FILE = "RoadmapResults_2017.xlsx"
+
+#: Historical years from the Roadmap model.
+HIST_YEARS = [2000, 2005, 2010, 2015]
+
+#: All Roadmap model timesteps.
+ALL_YEARS = [2000, 2005, 2010, 2015, 2020, 2025, 2030, 2035, 2040, 2045, 2050]
+
+COLUMN_MAP = {
+    "Mode": "Mode/vehicle type",
+}
+
+MODE_MAP = {
+    "Buses": "Bus",
+    "Freight Rail": "Freight trains",
+    "Aviation": "Domestic passenger airplanes",
+    "LDV": "Cars/light trucks",
+    "Passenger Rail": "Passenger trains",
+    "LHDT": "Freight trucks",
+    "MHDT_HHDT": "Freight trucks",
+}
+
+VAR_MAP = dict(
+    # Map of ICCT's Roadmap variables to IEA's EEI variable names
+    Stock_million="Stock (10^6 vehicle)",
+    Sales_million="New vehicle sales (10^6 vehicle)",
+    VKT_billion="Vehicle kilometre per year (10^6 vkm/yr)",
+    TKM_billion="Tonne kilometre per year (10^6 tkm/yr)",
+    PKM_billion="Passenger kilometre per year (10^6 pkm/yr)",
+    # TODO IEA EEI's is in PJ, adjust those values in future plots
+    Energy_PJ="Total final energy (EJ/yr)",
+    TTW_CO2_Mt="Total final emissions (MtCO2/yr)",
+    # TODO add PM2.5 to IAMconsortium/units: https://github.com/IAMconsortium/units
+    TTW_PM2_5_kt="Total final PM2.5 emissions (ktPM2.5/yr)",
+)
+
+UNITS = dict(
+    # Appearing in input file
+    Stock_million=(None, "megavehicle", "megavehicle"),
+    Sales_million=(None, "megavehicle", "megavehicle"),
+    VKT_billion=(None, "gigavehicle kilometer / year", "gigavehicle kilometer / year"),
+    TKM_billion=(None, "gigatonne kilometre / year", "gigatonne kilometre / year"),
+    PKM_billion=(
+        None,
+        "gigapassenger kilometre / year",
+        "gigapassenger kilometre / year",
+    ),
+    Energy_PJ=(None, "PJ", "EJ"),
+    TTW_CO2_Mt=(None, "Megatonne CO2", "Megatonne CO2"),
+    # TODO add PM2.5 to IAMconsortium/units: https://github.com/IAMconsortium/units
+    # TTW_PM2_5_kt=(None, "kilotonne PM2.5", "kilotonne PM2.5"),
+    # occupancy=(None, "pkm / vkm", "pkm / vkm"),
+    # vkm_capita=(None, "vkm / capita", "vkm / capita"),
+    # energy_vkm=(None, "MJ / vkm", "MJ / vkm"),
+    # mileage=(1000.0, "vkm / vehicle", "vkm / vehicle"),
+    # tkm_capita=(1000.0, "tonne kilometre / capita", "tonne kilometre / capita"),
+    # energy_tkm=(None, "MJ / tkm", "MJ / tkm"),
+    # load_factor=(None, "tkm / vkm", "tkm / vkm"),
+    # # Created below
+    # activity=(None, None, "gigapassenger kilometre / year"),
+)
+
+
+def get_roadmap_data(context, region=("Africa", "R11_AFR"), years=None, plot=False):
```

Comment:
Since all these diffs are classified as huge by GitHub (i.e. don't render per default), I'm only looking at some files. Type hinting might be a general topic of possible improvements, but doesn't need to happen in this PR.

Reply:
Opened #210/